proposals/channel-jamming-mitigation.md

```diff
@@ -0,0 +1,164 @@
+parts, upfront fees to address quick jamming and reputation to address slow
+jamming [2].
+
+### Quick Jamming Mitigation
+Traffic that is part of a quick jamming attack mimics the behavior of honest 
+nodes' failed payments, which this makes it difficult to identify. A solution
+that addresses quick jamming therefore *must have a trivial impact on honest 
+traffic, and significant cost to malicious traffic*. A solution that fails to 
+do this will only magnify an attacker's ability to disrupt the network, as they
+can leverage the jamming mitigation itself to further degrade honest traffic.
+
+An upfront fee, paid regardless of the outcome of a payment attempt, is proposed
+to economically compensate nodes for providing traffic with access to liquidity
+and slots. Simulations in [2] show that a fee of as little as 1% of the success
+case fees are sufficient to compensate the opportunity cost of forwarding nodes.
+
+Nodes that advertise `option_upfront_fee` (56/57) will advertise a new TLV in
+their `channel_update` which expresses unconditional fees as a percentage of 
+their success-case fees: 
+```
+tlv_stream: channel_update_tlvs
+types: 
+  type: 1 (upfront_fee)
+  data: 
+    [u32:fee_base_ppm]
+    [u32: fee_proportional_ppm]
+```
+TODO: maybe we shouldn't couple these with existing fees? What did we decide
+about the zero fee case?
+
+To save the network the bandwidth and storage required to transmit and save
+defaults, nodes that advertise `option_upfront_fee` in their node announcement
+should be assumed to have a base and proportional fee that is 1% of its success
+case fees. Nodes also will not relay and senders will not utilize channels with
+a `channel_update` messages where upfront fees are > 10% of their success-case
+equivalent. This eliminates the risk of nodes setting these fees so high that
+it becomes economically rational for forwarding nodes to fail payments for the
+sake of their upfront fees. 
+
+As with success-case fees, upfront fees are charged on the outgoing link and 
+accumulated on the incoming link when payments are accepted for forwarding. The
+`update_add_htlc` message is extended with a new TLV which specifies the amount
+that must be pushed to the remote peer on addition of the incoming HTLC to its
+channel state. 
+```
+tlv_stream: update_add_htlc_tlvs
+types: 
+  type: 2
+  data: 
+    [u64: upfront_fee_msat]
+```
+
+This field is mirrored with a TLV in the onion's payload so that the receiving
+peer can verify that the amount that they should receive (as is the case with 
+amount and expiry values at present). 
+```
+tlv_stream: payload
+types: 
+  type: 18 (upfront_fee)
+  data: 
+    [u64: upfront_fee_msat]
+```
+
+The upfront fee amount should simply be assigned to the receiving party's 
+balance when an incoming HTLC is added. These amounts are not enforceable 
+on-chain (they are dust), so there is no need to include an output for them in 
```

Comment:
Can they add up to something that is enforceable on chain?

Reply:
Theoretically, they need to hit ~330 sat to be non-dust. But we need to be able to make outputs per-hop, so probably safe to just assume it away?

Reply:
Is it true that after enough upfront fees are paid, it adds up to an amount that is enforceable on chain?

Reply:
If each hop had at least 330 sat of upfront fees then it would be enforceable.